jetty-ee10/jetty-ee10-quickstart/src/main/java/org/eclipse/jetty/ee10/quickstart/QuickStartGeneratorConfiguration.java

```diff
@@ -167,7 +168,11 @@ public void generateQuickStartWebXml(WebAppContext context, OutputStream stream)
         // Set some special context parameters
 
         // The location of the war file on disk
-        AttributeNormalizer normalizer = new AttributeNormalizer(context.getBaseResource());
+        Resource base = (Resource)context.getAttribute(WebInfConfiguration.UNPACKED_RESOURCE_BASE);
+        //Not unpacked, pick the first base
+        if (base == null && context.getBaseResource() != null)
+            base = context.getBaseResource().iterator().next();
+        AttributeNormalizer normalizer = new AttributeNormalizer(base);
```

Comment:
On consideration, I've changed my mind and I think that taking the first resource is always the best we can do.  In the case that `UNPACKED_RESOURCE_BASE` is set, is the well known case that the base resource will be well known and taking the first will work.     There are still strange edge cases if somebody sets a strange/complex baseResource, but the attribute will not help with those anyway.

Reply:
What happens if there is no base resource?

Reply:
In the strange/complex baseResource case, couldn't the user specify/override for a simpler `UNPACKED_RESOURCE_BASE` attribute to be precise on behavior?